File: frontend/src/types/interviewPrep.ts
export interface InterviewPrep {
  id: number;
  project_id: number;
  type: string;
  project_pitch?: string | null;
  technical_explanation?: string | null;
  resume_bullets?: string | null;
  debugging_story?: string | null;
  architecture_explanation?: string | null;
  star_answer?: string | null;
  created_at: string;
}

export interface InterviewPrepGenerateResponse {
  prep: InterviewPrep;
}

File: backend/app/schemas/interview_prep.py
from datetime import datetime

from pydantic import BaseModel


class InterviewPrepRead(BaseModel):
    id: int
    project_id: int
    type: str
    project_pitch: str | None = None
    technical_explanation: str | None = None
    resume_bullets: str | None = None
    debugging_story: str | None = None
    architecture_explanation: str | None = None
    star_answer: str | None = None
    created_at: datetime

    class Config:
        from_attributes = True


class InterviewPrepGenerateRequest(BaseModel):
    focus: str | None = None


class InterviewPrepGenerateResponse(BaseModel):
    prep: InterviewPrepRead
